### PR TITLE
Improve shiny server UI

### LINKED ISSUE #12: Improve server.R
There are current issues with R server:

- The "Load codelist" button UI doesn't provide functionality.
- It is too easy to return errors by entering a codelist which does not overlap with the selected codes and search terms.
- Selection of input should occur outside of the data filtering event.

### Implementation

Suggestion of implementation

```r
 codelist_codes <- eventReactive(input$load_codelist, {
    req(input$codelist_id)
    req(input$codelist_version_tag)
    url_base <- "https://www.opencodelists.org/codelist/"
    url_downlod_suffix <- "download.csv"
    url_codelist_csv <- paste0(url_base, input$codelist_id, "/", input$codelist_version_tag, "/", url_downlod_suffix)
    codelist_codes <- read.csv(url_codelist_csv, header = T)[,1]
    tryCatch({
      read.csv(input$url)
    }, error = function(e) {
      showNotification("Error loading URL data", type = "error")
      NULL
    })
  })
```  

### Completion:

- [ ] Restore functionality to Load Codelists button
- [ ] Implement input selection outside of the data filtering
- [ ] Display appropriate errors for different problems:

      1. Codelist entered in the wrong format
      2. No overlapping codes between the codelist and the seacrh term

- [ ] Document how to zoom in & out on plotly chart

### PR DESCRIPTION
**Pull request by milanwiedemann**
Closes #12

### CODE REVIEW

File: R/server.R [hunk 1]
```diff
@@ -1,110 +1,165 @@
...
-  
-  # Filtered data
+
+  # DATA: OpenCodelist
+  selected_codelist <- reactive({
+    req(input$codelist_slug)
+
+    tryCatch(
+      {
+        codelist_s7 <- get_codelist(input$codelist_slug)
```
Comment (atamborska) [inline R/server.R:46]: Great to have this processing outside of the data filter. How does R know `get_codelist` function? I can't see it defined elsewehre in the package.

File: R/ui.R [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            "Enter <codelist_id>/<version_id>, for example: 'nhsd-primary-care-domain-refsets/bp_cod/20200812'",
+            options = list(
+              customClass = "left-align-tooltip"
+            )
+          )
+        ),
+        textInput(
+          "codelist_slug",
+          "Codelist ID / Version Tag",
+          placeholder = "nhsd-primary-care-domain-refsets/bp_cod/20200812",
```
Comment (atamborska) [inline R/ui.R:59]: I still wonder if copying in the whole URL is not simplest from the user perspective (just one copy and paste), as doesn't require manually deleting the "https://" etc.

**Review by atamborska — APPROVED** [review]
Great work - thank you for all your efforts on this!

File: R/server.R (hunk 1 above)
Comment (milanwiedemann) [reply R/server.R:46]: It's part of the package, hidden at the bottom of `R/opencodelists.R`  https://github.com/ebmdatalab/codeusage/blob/bc4ffc18fefbcb27c2d1d22565f21d4e13f5b5c1/R/opencodelists.R#L28-L74

**Pull request merged**